javaslang/generator/Generator.scala

```diff
@@ -2473,7 +2447,15 @@ def generateMainClasses(): Unit = {
             public $wrapperType getAt(Object array, int index) { return cast(array)[index]; }
 
             @Override
-            public void setAt(Object array, int index, Object value) { cast(array)[index] =$cast value; }
+            public void setAt(Object array, int index, $wrapperType value) ${if (isPrimitive) "throws ClassCastException " else ""}{
+                ${if (isPrimitive)
+                """if (value == null) {
+                  |    throw new ClassCastException();
```

Comment:
My `xs"""..."""` implementation does _automatic_ indentation. No more `.stripMargin` necessary. But we leave it this way, it is ok!